src/ResourceManager/Advisor/Commands.Advisor/Cmdlets/Utilities/RecommendationHelper.cs

```diff
@@ -0,0 +1,196 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Advisor.Cmdlets.Utilities
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Management.Advisor.Models;
+    using Microsoft.Azure.Commands.Advisor.Cmdlets.Models;
+
+    /// <summary>
+    /// Recommendation helper class
+    /// </summary>
+    public static class RecommendationHelper
+    {
+        /// <summary>
+        /// Filter recommendations by given category and resourceGroup name.
+        /// </summary>
+        /// <param name="recListTobeFiltered">List to be filtered</param>
+        /// <param name="category">Category name</param>
+        /// <param name="resourceGroup">Resource group name</param>
+        /// <returns>Filtered list of recommendations</returns>
+        public static List<PsAzureAdvisorResourceRecommendationBase> ReccomendationFilterByCategoryAndResource(IEnumerable<PsAzureAdvisorResourceRecommendationBase> recListTobeFiltered, string category, string resourceGroup)
+        {
+            if (recListTobeFiltered == null || recListTobeFiltered.Count() == 0)
+            {
+                return null;
+            }
+
+            List<PsAzureAdvisorResourceRecommendationBase> filteredList = new List<PsAzureAdvisorResourceRecommendationBase>();
+
+            // Filter by category only if its a valid input
+            if (Category.Cost.Equals(category) || Category.HighAvailability.Equals(category) || Category.Performance.Equals(category) || Category.Security.Equals(category))
+            {
+                if (!string.IsNullOrEmpty(category) && !string.IsNullOrEmpty(resourceGroup))
```

Comment:
Why do you need to check is category is null or empty? You just did a check to make sure it was one of four specific values above.